### ISSUE TITLE
Streams

### ISSUE THREAD

**Issue by hdf** [node-ffi/node-ffi#276]
Is it possible (or will it ever be) to give a stream as parameter for the dll function? My function in the dll returns a lot of text data line by line streaming it, how do I process that on the node side? I don't think creating a 50 Megabyte buffer would be the best solution... Especially if both the dll and node create a version of it...

**Comment by TooTallNate** [comment]
Most likely you would have to implement the `stream.Readable` on top of node-ffi and your dll function, but can you show me some example code of what you're talking about so I can get a better idea?

**Comment by hdf** [comment]
My dll uses openmp to crunch a lot of math. I'm trying to build a nodejs based distribution system for data packages. The client downloads the dll, gets data package from the server, than processes it, results would be written to a file, if this was not a dll, but instead of a file. I would like to stream it back to the node client, so that it can than send it back to the package server. I have made an asm.js version, that I was running in webworker-threads, but for some reason it is slow. The asm.js version does in 8 seconds what the the c++ version does in 1.6. Both use all CPU cores.

**Comment by TooTallNate** [comment]
Do you invoke a function on the DLL multiple times and it returns one line at a time?

**Comment by hdf** [comment]
No. As mentioned, the function in the dll gets called once, and it pumps back a lot of data line by line. By "pumps" I mean fprintf -s a char array in to a stream. (Currently file stream.)
